lnrpc/invoicesrpc/invoices.proto

```diff
@@ -55,10 +55,19 @@ service Invoices {
     rpc SettleInvoice (SettleInvoiceMsg) returns (SettleInvoiceResp);
 
     /*
-    LookupInvoiceV2 attempts to look up at invoice. An invoice can be refrenced
+    LookupInvoiceV2 attempts to look up at invoice. An invoice can be referenced
     using either its payment hash, payment address, or set ID.
     */
     rpc LookupInvoiceV2 (LookupInvoiceMsg) returns (lnrpc.Invoice);
+
+    /*
+    HtlcModifier is a bidirectional streaming RPC that allows a client to
+    intercept and modify the HTLCs that attempt to settle the given invoice. The
+    server will send HTLCs of invoices to the client and the client can modify
+    some aspects of the HTLC in order to pass the invoice acceptance tests.
+    */
+    rpc HtlcModifier (stream HtlcModifyResponse)
```

Comment:
Hmm, could perhaps be useful in the future. Not sure if we need to optimize prematurely. Since this will be called way less often than the forwarding HTLC interceptor for example.